R/shinyGadget.R

```diff
@@ -1,69 +1,164 @@
 projectGadget <- function() {
+
+  if (!requireNamespace("shinyFiles", quietly = TRUE)) {
+    stop("The shinyFiles package is required for this gadget to work. Please install it.",
+         call. = FALSE)
+  }
+
   ui <- miniUI::miniPage(
     miniUI::gadgetTitleBar("Create pRoject"),
     miniUI::miniContentPanel(
+      shiny::tags$style("#folder {width: 435px; };",
+        type = "text/css"
+      ),
       shiny::inputPanel(
         shiny::selectInput(
           "projectType",
           label = "Project Type",
-          choices = c("Basic", "Analysis", "Package", "Training"),
+          choices = c("Basic", "Package", "Analysis", "Training"),
           selected = "Basic"
         ),
-        shiny::textInput("name", "Project Name")
+        shiny::selectInput(
+          "initial_status",
+          label = "Initial Status",
+          choices = statuses()$status,
+          selected = "wip"
+        ),
+        shiny::textInput("name", "Project Name", value = NULL),
+        shiny::textInput("title", "Project Title", value = NULL),
+        shiny::div("Project Folder", class = "control-label", style = "font-weight: bold;"),
+        shiny::br(),
+        shiny::br(),
+        shiny::br(),
+        shinyFiles::shinyDirButton("folder",
+          label = "Choose Folder",
+          title = "Please select a folder",
+          class = "btn-primary"
+        ),
+        shiny::br(),
+        shiny::uiOutput("folder")
       ),
       shiny::uiOutput("ui"),
       shiny::conditionalPanel(
-        condition = "input.projectType != 'Package'",
+        condition = "input.externalSetup",
         shiny::inputPanel(
-          shiny::h4("Best Practices"),
-          shiny::br(),
-          shiny::checkboxInput("git", "Use Git", value = TRUE),
-          shiny::checkboxInput("travis", "Use Travis", value = TRUE),
-          shiny::selectInput("packagedeps",
-            label = "Use a package for handling reproducibility",
-            choices = c("none", "packrat", "checkpoint"),
-            selected = "packrat"
-          )
+          shiny::radioButtons("git_service",
+                              label = "Git Service",
+                              choices = "GitHub"
+          ),
+          shiny::radioButtons("private", label = "Repository",
+            choices = c("public" = FALSE, "private" = TRUE)
+          ),
+          shiny::radioButtons("protocol",
+            label = "Protocol",
+            choices = c("ssh", "http"),
+            select = "ssh"
+          ),
+          shiny::radioButtons("ci_activation",
+            label = "Continuous Integration",
+            choices = c("none"="", "travis")
+          ),
+          shiny::textInput("login", label = "Login", value = gh::gh_whoami()$login)
         )
       )
     )
   )
 
   server <- function(input, output, session) {
+    volumes <- c(
```

Comment:
could one of them (maybe even the first one) be `getwd()` well passed when calling the app?
I have that in https://github.com/ropenscilabs/blogyaml/blob/master/inst/shiny/app.R where `.post.path` is set before calling the app https://github.com/ropenscilabs/blogyaml/blob/master/R/edit_tags.R#L14

Reply:
Cool - will add this 👍

Reply:
it looks great!